src/borg/_hashindex.c

```diff
@@ -134,20 +145,21 @@ hashindex_lookup(HashIndex *index, const void *key, int *skip_hint)
     int idx = start;
     int num_buckets = index->num_buckets;
     int offset;
-    for(offset=0; ; offset++) {
-      /* lookups ++;  //TRACE_PERF */
+    for(offset=0; ;offset++) {
+        if (DEBUG) {
+            lookups ++;
+        }
```

Comment:
hmm, you seem to rely on the compiler optimizing away this debug code. one could also use #ifdef DEBUG to let the precompiler just remove all the code.